src/report/UnityMovementReport.java

```diff
@@ -0,0 +1,175 @@
+
+class NodeInformation {
+    HashMap<String, Double> locations = new HashMap<>();
+    double lastTimeStep = 0;
+    String currentNodeState = new QueueState().getStateName();
+
+    NodeInformation(double lastTimeStep) {
+        this.lastTimeStep = lastTimeStep;
+    }
+}
+
+public class UnityMovementReport extends Report implements MovementListener {
+
+    private HashMap<Integer, NodeInformation> nodeMap = new HashMap<>();
+    private static final NodeState[] states = new NodeState[]{
+            new QueueState(),
+            new EntryState(),
+            new WardrobeState(),
+            new MainStageState(),
+            new OutdoorAreaState(),
+            new MetalBunkerState(),
+            new TechnoBunkerState(),
+            new CocktailBarState(),
+            new BeerBarState(),
+            new ShotBarState(),
+            new ShishaBarState(),
+            new PizzaBarState(),
+            new SideWCState(),
+            new MainWCState(),
+            new ExitState(),
+            new WardrobeBeforeLeavingState()
+    };
+
+    @Override
+    public void newDestination(DTNHost host, Coord destination, double speed) {
+        NodeInformation currentNodeInfo = nodeMap.get(host.getAddress());
+        HashMap<String, Double> locationHashMap = currentNodeInfo.locations;
+
+        Double currentTime = getSimTime();
+        Double time = currentTime - currentNodeInfo.lastTimeStep;
+        Double count = locationHashMap.get(currentNodeInfo.currentNodeState);
+        if (count == null) {
+            locationHashMap.put(currentNodeInfo.currentNodeState, time);
+        } else {
+            locationHashMap.put(currentNodeInfo.currentNodeState, count + time);
+        }
+
+        try {
+            StatefulRwp movModel = (StatefulRwp) host.movement;
+            String labelLocation = movModel.lastState.getStateName();
+            currentNodeInfo.lastTimeStep = currentTime;
+            currentNodeInfo.currentNodeState = labelLocation;
+            return;
+        } catch (Exception e) {
+            // Do nothing
+        }
+
+        currentNodeInfo.lastTimeStep = currentTime;
+        currentNodeInfo.currentNodeState = "Other";
+    }
+
+    @Override
+    public void initialLocation(DTNHost host, Coord location) {
+        nodeMap.put(host.getAddress(), new NodeInformation(getSimTime()));
+    }
+
+    @Override
+    public void done() {
+        //write summary
+        write("---- DONE ----");
+        //HashMap.SimpleEntry<Integer, InfectionData>
+        for(Map.Entry<Integer, NodeInformation> entry : nodeMap.entrySet()){
+            String locations = "";
+            int total = 0;
+            for(NodeState state : states) {
+                Double value = entry.getValue().locations.get(state.getStateName());
+                if (value == null) {
+                    value = 0.0;
+                }
+                total += value;
+                locations += state.getStateName() + ": " + value + "   |   ";
```

Comment:
```suggestion
                locations += state.getStateName() + ": " + value + ",";
```
Ich würde Komma zum Splitten verwenden, dann ist die Zeile etwas kürzer und trotzdem noch gut lesbar.

Reply:
Einen Tab mit \t fände ich auch gut, dann ist der Report deutlich leserlicher

Reply:
Wie genau meint ihrs?